client/templates/layouts/default.js

```diff
@@ -1,30 +1,27 @@
-const handleRedirect = ( routes, redirect ) => {
-	let currentRoute = FlowRouter.getRouteName();
-	if ( routes.indexOf( currentRoute ) > -1 ) {
-		FlowRouter.go( redirect );
-		return true;
-	}
+const authenticated = () => {
+  return !Meteor.loggingIn() && Meteor.user();
+}
+
+const handleRedirect = (currentRoute) => {
+  let authData = currentRoute.group.options.authentication;
+  if (authData.requireLoggedIn ^ authenticated()) {
```

Comment:
requireLoggedIn XOR authenticated: Only redirect if either one is true; Not both true or both false.